Google.Api.Gax.Grpc/ApiCall.cs

```diff
@@ -185,5 +186,24 @@ public sealed class ApiCall<TRequest, TResponse>
             GaxPreconditions.CheckNotNull(valueSelector, nameof(valueSelector));
             return WithCallSettingsOverlay(request => CallSettings.FromGoogleRequestParamsHeader(parameterName, valueSelector(request)));
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="extractor"></param>
+        /// <returns></returns>
+        public ApiCall<TRequest, TResponse> WithExtractedGoogleRequestParam(HeaderParameterExtractor<TRequest> extractor)
```

Comment:
We might want to consider moving the body of the lambda expression to HeaderParameterExtractor, to avoid the duplication between this and ApiServerStreamingCall (and others?). That can be a later refactor though.

Reply:
This one I am happy to leave to the later refactor. I would like to separate combining the header parameter from extracting them as well, and I was not able to get a clean version that I would like there.